### PR TITLE
Feat: DS stepper component

### PR DESCRIPTION
**Pull request by josenriagu**
## Description

- create stepper component for stepped actions
- add and update stories

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

**Comment by quininez** [comment]
From the design I can see that the component has 3 states for each step: a grey indicator when the step is not reached, a filled accent circle for a completed step and an empty circle for the current step. From what I can see in the implementation the component now only has 2 states.

File: libs/design-system-components/src/stories/dscomponents/stepper.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import type { Meta, StoryObj } from '@storybook/react';
+import Stepper, { StepperProps } from '../../components/Stepper';
+
+Stepper.displayName = 'Spinner';
```
Comment (SeverS) [inline libs/design-system-components/src/stories/dscomponents/stepper.stories.tsx]: stepper's display name is spinner?

Comment (josenriagu) [reply libs/design-system-components/src/stories/dscomponents/stepper.stories.tsx]: fixing this now

**Comment by josenriagu** [comment]
> From the design I can see that the component has 3 states for each step: a grey indicator when the step is not reached, a filled accent circle for a completed step and an empty circle for the current step. From what I can see in the implementation the component now only has 2 states.

Screenshot 2024-08-05 at 09 03 29
This is what it looks like on the story.

**Review by quininez — APPROVED** [review]

**Review by SeverS — APPROVED** [review]

**Pull request merged**